dependency/health_service_test.go

```diff
@@ -675,6 +691,136 @@ func TestHealthServiceQuery_Fetch(t *testing.T) {
 	}
 }
 
+// TestHealthServiceQuery_Fetch_SamenessGroup tests fetching services with sameness group
+// and asserts that failover works when a service in the sameness group fails and consul-template
+// reacts to the change.
+func TestHealthServiceQuery_Fetch_SamenessGroup(t *testing.T) {
```

Comment:
There's some behavior I don't think we're currently testing:
* Querying for a service in a sameness group in a partition that the SG isn't defined in returns nothing (or an error?).
* If a service becomes healthy again in partitionOne, it returns partitionOne again
* If a second service instance comes up, it is also returned.
* If a service instance is deregistered it is no longer retured.
* If a sameness group is deleted it returns an error (what happens if it gets recreated?)
* If a sameness group ordering or list of members changes the returned service instance is updated
* This works across peers (i.e. multiple DCs)

Maybe some of those tests are too difficult to write or don't make sense because they're double-testing things? Interested in your thoughts here.

Reply:
I'll cross reference the blocking query tests in consul, but I added most of this there so that this was just a test to make sure that consul-template hooks into what the api is doing.

Reply:
Ahh understood. Then for this test can you add a comment to the effect of:

This test ensures that consul-template re-runs when the blocking query updates. The different behaviors of the blocking query, e.g. "when a service becomes unhealthy it should failover, when it becomes healthy again it should fail back", are tested in the Consul codebase.

Reply:
Will do.  I will cross-reference your suggestions above and in the other comment and open a PR for the missing scenarios in consul.

Reply:
updated test docstring in 7c90b52ed98516de357533004645677e66074442.  will resolve this when a PR for missing behavior assertions is opened in Consul.

Reply:
@lkysow thank you for calling these out.  In adding coverage I found a bug when the sameness groups does not exist or is deleted.  New tests and bug fix are in this PR: https://github.com/hashicorp/consul-enterprise/pull/9445

Here is a summary of what you asked about and confirmation each is in the `TestHealth_ServiceNodes_BlockingQuery_SamenessGroupParam` in consul enterprise after additional tests (and some already existed)
- If a service becomes healthy again in partitionOne, it returns partitionOne again
- If a service instance is deregistered it is no longer returned.
- If a sameness group ordering or list of members changes the returned service instance is updated
- If a second service instance comes up, it is also returned.
- If a sameness group is deleted it returns an error (what happens if it gets recreated?)
- Querying for a service in a sameness group in a partition that the SG isn't defined in returns an error.
- Querying for a Sameness group that does not exist returns an error (should be same code flow as above but worth having a test).